lib/event.ts

```diff
@@ -0,0 +1,28 @@
+import { LogDescription } from '@ethersproject/abi';
+import { ContractReceipt } from 'ethers';
+import { Interface } from 'ethers/lib/utils';
+
+export function getLogs(
+  contractInterface: Interface,
+  receipt: ContractReceipt,
+  only: string[] = [],
+) {
+  const events: LogDescription[] = [];
+
+  for (const log of receipt.logs) {
+    try {
+      let description = contractInterface.parseLog(log);
+
+      if (only.length > 0 && !only.includes(description.name)) {
+        continue;
+      }
+
+      events.push(contractInterface.parseLog(log));
```

Comment:
Already parsed, so just push `description`.
```suggestion
      events.push(description);
```